tty.c

```diff
@@ -552,3 +582,191 @@ void tty_child(int socket)
 		close(sfd);
 	}
 }
+
+struct valmap {
+	const char *name;
+	void (*fn)(struct tty_opts *, const char *, const char *, const void *);
+	const void *cookie;
+};
+
+static void parse_flag(struct tty_opts *opts, const char *key, const char *val, const void *cookie)
+{
+	if (val != NULL) {
+		errx(2, "tty option '%s' must have no value", key);
+	}
+
+	const struct termios *tios = cookie;
+	struct termios *dest = &opts->termios;
+	if (key[0] == '-') {
+		dest = &opts->neg_termios;
+	}
+
+	dest->c_iflag |= tios->c_iflag;
+	dest->c_oflag |= tios->c_oflag;
+	dest->c_lflag |= tios->c_cflag;
+	dest->c_lflag |= tios->c_lflag;
+}
+
+static void parse_cc(struct tty_opts *opts, const char *key, const char *val, const void *cookie)
+{
+	if (key[0] == '-') {
```

Comment:
Should we support disabling the control character (setting to _POSIX_VDISABLE)?  If so then the leading - might make sense.